[shader-slang] support find_package

FrostyLeaves opened the pull request:
When building shader-slang via vcpkg, the shader-slangConfig.cmake file is automatically generated to add the slang library, allowing you to quickly add slang to your project in CMake using find_package.

In accordance with the contribution guidelines, added the unofficial prefix and incremented the port-version.


- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

FrostyLeaves commented:
> @microsoft-github-policy-service agree

@microsoft-github-policy-service agree

LilyWangLL commented on ports/shader-slang/portfile.cmake:
```diff
@@ -115,6 +115,42 @@ vcpkg_copy_tools(TOOL_NAMES slangc slangd SEARCH_DIR "${BINDIST_PATH}/bin")
 file(GLOB headers "${BINDIST_PATH}/include/*.h")
 file(INSTALL ${headers} DESTINATION "${CURRENT_PACKAGES_DIR}/include")
 
+set(SLANG_CONFIG_CONTENT "
+add_library(unofficial::slang::slang SHARED IMPORTED)
+
+# Compute the installation prefix relative to this file.
+get_filename_component(_IMPORT_PREFIX \"\${CMAKE_CURRENT_LIST_FILE}\" PATH)
+get_filename_component(_IMPORT_PREFIX \"\${_IMPORT_PREFIX}\" PATH)
+get_filename_component(_IMPORT_PREFIX \"\${_IMPORT_PREFIX}\" PATH)
+if(_IMPORT_PREFIX STREQUAL \"/\")
+  set(_IMPORT_PREFIX \"\")
+endif()
+
+file(GLOB libs \"\${_IMPORT_PREFIX}/lib/slang.lib\")
+
+file(GLOB dyn_libs
+  \"\${_IMPORT_PREFIX}/lib/libslang.dylib\"
+  \"\${_IMPORT_PREFIX}/lib/libslang.so\"
+  \"\${_IMPORT_PREFIX}/lib/slang.dll\"
+) ")
+
+string(APPEND SLANG_CONFIG_CONTENT "
+
+set_target_properties(unofficial::slang::slang PROPERTIES
+  INTERFACE_INCLUDE_DIRECTORIES \"\${_IMPORT_PREFIX}/include\"
+  IMPORTED_LOCATION \"\${dyn_libs}\" ")
+
+if (VCPKG_TARGET_IS_WINDOWS)
+  string(APPEND SLANG_CONFIG_CONTENT "
+  IMPORTED_IMPLIB \"\${libs}\"  ")
+endif()
+
+string(APPEND SLANG_CONFIG_CONTENT "
+)
+")
+
```
Could you please add a new file `unofficial-shader-slangConfig.cmake` in this path to add these contents in it? Don't add these in `portfile.cmake`.

dg0yt commented:
The vcpkg port downloads binary packages. These packages contain official CMake config, but it isn't installed at the moment. So the actual task is to install and fixup the official config IMO. Alternatively, reimplement the same interface instead of creating an unofficial one.

FrostyLeaves committed:
[shader-slang] Update to 2025.6.4

Update to 2025.6.4.

install slangConfig.cmake

add supports !(static & staticcrt)

run x-add-version

FrostyLeaves commented:
> This vcpkg port downloads binary packages. These packages contain official CMake config, but it isn't installed at the moment. So the actual task is to install and fixup the official config IMO. Alternatively, reimplement the same interface instead of creating an unofficial one.

I submitted a fix for the official config: https://github.com/shader-slang/slang/pull/6679

This pull request now uses the official CMake config from Slang (after the corresponding Slang version is released).

JohannesBeranek commented:
@FrostyLeaves looks like the new version has been released (2025.6.4)
Been waiting for this for some time now, so I thought I'd ping you :)

Thanks for the work!

FrostyLeaves commented:
It seems like an expected-to-fail CI check has failed.

![image](https://github.com/user-attachments/assets/d789f7b0-6fd2-4972-a6c5-58fb15d5d96b)
![image](https://github.com/user-attachments/assets/ec6745b6-0887-4887-bc13-cb74f713daf5)

FrostyLeaves commented:
> @FrostyLeaves looks like the new version has been released (2025.6.4) Been waiting for this for some time now, so I thought I'd ping you :)
> 
> Thanks for the work!

Using the official CMake config turned out to be more troublesome than I expected.
Although the changes make it work, I'm not sure my commits would be allowed for submission.

dg0yt commented on ports/shader-slang/portfile.cmake:
```diff
@@ -75,6 +75,26 @@ if(VCPKG_SHADER_SLANG_UPDATE)
+)
+
+# install cmake files
+set(SLANG_INSTALL_CMAKE_PATH ${CURRENT_PACKAGES_DIR}/share/slang)
+file(INSTALL ${slang_cmake_configs} DESTINATION ${SLANG_INSTALL_CMAKE_PATH})
```
```suggestion
set(SLANG_INSTALL_CMAKE_PATH "${CURRENT_PACKAGES_DIR}/share/slang")
file(INSTALL "${slang_cmake_configs}" DESTINATION "${SLANG_INSTALL_CMAKE_PATH}")
```

dg0yt commented on ports/shader-slang/portfile.cmake:
```diff
@@ -75,6 +75,26 @@ if(VCPKG_SHADER_SLANG_UPDATE)
+
+# install cmake files
+set(SLANG_INSTALL_CMAKE_PATH "${CURRENT_PACKAGES_DIR}/share/slang")
+file(INSTALL "${slang_cmake_configs}" DESTINATION "${SLANG_INSTALL_CMAKE_PATH}")
```
I am sorry, I failed on the result variable of the glob :facepalm:
```suggestion
file(INSTALL ${slang_cmake_configs} DESTINATION "${SLANG_INSTALL_CMAKE_PATH}")
```

FrostyLeaves replied:
resolved

LilyWangLL commented:
The usage test failed due to the target missing include property. 

And the installed `share` directory only contains release target files and is missing debug target files.
![image](https://github.com/user-attachments/assets/b892f256-95b0-4aeb-b5d0-6402a307f4ff)
